neuralop/models/base_model.py

```diff
@@ -76,7 +76,33 @@ def __new__(cls, *args, **kwargs):
         instance._init_kwargs = kwargs
 
         return instance
-    
+
+    def state_dict(self, **kwargs):
+        """
+        state_dict subclasses nn.Module.state_dict() and adds a metadata field
+        to track the model version and ensure only compatible saves are loaded.
+        """
+        print("In state dict")
+        state_dict = super().state_dict(**kwargs)
+        state_dict.update({'neuralop_metadata': self._init_kwargs})
+        return state_dict
+
+    def load_state_dict(self, state_dict, **kwargs):
+        """
+        state_dict subclasses nn.Module.load_state_dict() and adds a metadata field
+        to track the model version and ensure only compatible saves are loaded.
+        """
+        metadata = state_dict.get('neuralop_metadata')
+        if metadata:
+            saved_version = metadata['_version']
+            self_version = self._init_kwargs['_version']
```

Comment:
Let's use get('_version', None) and add another warning if the model doesn't have a version attached.